Products/ZenMessaging/queuemessaging/publisher.py

```diff
@@ -31,6 +31,8 @@
 log = logging.getLogger('zen.queuepublisher')
 
 MODEL_TYPE = ProtobufEnum(modelevents_pb2.ModelEvent, 'model_type')
+BEFORE_COMMIT_HOOK_TIMER = Metrology.utilization_timer("zen.queuepublisher.beforeCompletionHookTimer")
```

Comment:
I don't think these meters should be created in the module's global scope. This leads to them being created in process that don't even have access to ZODB. See what happens when I put a pdb.set_trace() on the same line, then run zenpython.

```
(zenoss) [zenoss@5b5f49ae7072 ~]$ zenpython run
> /opt/zenoss/Products/ZenMessaging/queuemessaging/publisher.py(40)<module>()
-> class ModelChangePublisher(object):
(Pdb)
```

If you want them in the module's scope, you could set them to None here, then initialize them on demand when they're used.

Reply:
Added the Metrology timers as a decorator to the appropriate methods.
The Metrology package has his own registry package so we only can use it without writing additional getter function.